### PR TITLE
front: refacto simulationResultsHelpers TS and RTK

### LINKED ISSUE #4951: front: migrate RTK endpoint in simulationResultsHelpers and migrate to TS
### Description and goal

Use RTK for endpoints calls

### Acceptance criteria

The endpoints are all called with RTK
File written in ts

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close #4951

### CODE REVIEW

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          path_id: details.path_id || trainDetail?.path_id,
+          rolling_stock_id: details.rolling_stock_id || trainDetail?.rolling_stock_id,
+          train_name: details.train_name || trainDetail?.train_name,
+        };
+        dispatch(updateReloadTimetable(true));
+        dispatch(osrdEditoastApi.endpoints.patchTrainSchedule.initiate({ body: [trainSchedule] }));
+        dispatch(updateReloadTimetable(false));
+      } catch (e) {
+        dispatch(updateReloadTimetable(false));
+        if (e instanceof Error) {
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts]: Not sure if it is better, but if we want to use rtk fully, you could use the "isError" & "error" objects from rtk ?
(Not tested but maybe by removing de try/catch and put an if (isError && error) / else)

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          dispatch(
+            setFailure({
+              name: e.name,
+              message: e.message,
+            })
+          );
+        }
+      }
+    } catch (e) {
+      if (e instanceof Error) {
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts]: same here

Comment (Uriel-Sautron) [reply front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts]: Done

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts (hunk 1 above)
Comment (Uriel-Sautron) [reply front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.ts]: Done

**Review by SharglutDev — APPROVED** [review]
LGTM :)

**Review by nicolaswurtz — APPROVED** [review]

**Pull request merged**